File: project/client/src/app/model/Project.ts
import { IBaseEntity } from "../core/model/BaseEntity";
import { ITimeLog } from "./TimeLog";
import { ICompany } from "./Company";

export interface IProject extends IBaseEntity {
    name: string;
    deadline: string;
    companyId: string;
    completed: boolean;
}

export interface IProjectDto extends IBaseEntity {
    name: string;
    deadline: Date | null;
    completed: boolean;
    company?: ICompany;
    logs: ITimeLog[];

    totalSpentTime: number;
    isEnded: boolean;
}

File: project/server/Domain/Project.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Project
    {
        public Guid Id { get; set; }
        public DateTime Deadline { get; set; }
        public Guid CompanyId { get; set; }
        public string Name { get; set; }
        public bool Completed { get; set; }
        public virtual Company Company { get; set; }
        public virtual List<TimeLog> Logs { get; set; }
    }
}
